src/rules/prefer-repository-shorthand.ts

```diff
@@ -0,0 +1,103 @@
+import type ESTree from "estree";
+import type { JSONLiteral } from "jsonc-eslint-parser/lib/parser/ast.js";
+
+import { createRule } from "../createRule.js";
+
+const isGitHubUrl = (url: string) =>
+	/^(?:git\+)?(?:ssh:\/\/git@|http?s:\/\/)?(?:www\.)?github\.com\/.*\/.*/.test(
+		url,
+	);
+
+const cleanGitHubUrl = (url: string) =>
```

Comment:
[Refactor] The `isCleanGitHubUrl` function above already performs a RegExp run on the string. In theory the relevant `(?:...)` non-capturing groups could be turned into `(...)` capturing ones, which would then be used in constructing the shorthand. Thoughts?

Reply:
This wasn't resolved? The code is still doing two regex executions.

Tbh it's fine though, the URLs aren't so big and fancy that the perf impact would be noticeable.